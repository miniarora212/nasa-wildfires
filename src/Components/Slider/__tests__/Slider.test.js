import { cleanup, render } from "@testing-library/react";
import React from "react"
import moment from 'moment';
import RangeSlider from "../Slider";
import "@testing-library/jest-dom/extend-expect";

afterEach(cleanup);

describe('Slider', () => {
  
    const defaultProps = {
        min: 2,
        max: 24,
        defaultVal: 12,
        setValues: jest.fn(),
        formatter: jest.fn()
    };

    it('Should render the slider successfully', () =>{
        const { asFragment } = render(<RangeSlider {...defaultProps} />);
        expect(asFragment(<RangeSlider {...defaultProps}/>)).toMatchSnapshot();
    });
});