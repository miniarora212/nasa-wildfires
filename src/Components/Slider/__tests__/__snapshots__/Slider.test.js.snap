// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider Should render the slider successfully 1`] = `
<DocumentFragment>
  <div
    class="sliderContainer w-50"
  >
    <div
      class="ant-slider ant-slider-with-marks"
    >
      <div
        class="ant-slider-rail"
      />
      <div
        class="ant-slider-track"
        style="left: 0%; width: 45.45454545454545%;"
      />
      <div
        class="ant-slider-step"
      >
        <span
          class="ant-slider-dot ant-slider-dot-active"
          style="left: 0%;"
        />
        <span
          class="ant-slider-dot"
          style="left: 100%;"
        />
      </div>
      <div
        aria-disabled="false"
        aria-valuemax="24"
        aria-valuemin="2"
        aria-valuenow="12"
        class="ant-slider-handle"
        role="slider"
        style="left: 45.45454545454545%; transform: translateX(-50%);"
        tabindex="0"
      />
      <div
        class="ant-slider-mark"
      >
        <span
          class="ant-slider-mark-text ant-slider-mark-text-active"
          style="transform: translateX(-50%); left: 0%;"
        >
          Start
        </span>
        <span
          class="ant-slider-mark-text"
          style="transform: translateX(-50%); left: 100%;"
        >
          End
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
